plugins/civo/api_key.go

```diff
@@ -0,0 +1,105 @@
+package civo
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://www.civo.com/docs/account/api-keys"),
+		ManagementURL: sdk.URL("https://dashboard.civo.com/security"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Civo.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 50,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "API Name to identify the API Key.",
+			},
+			{
+				Name:                fieldname.DefaultRegion,
+				MarkdownDescription: "The default region to use for this API Key.",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryCivoConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"CIVO_API_KEY_NAME": fieldname.APIKeyID,
+	"CIVO_API_KEY":      fieldname.APIKey,
+}
```

Comment:
I could only find these envvars documented for `civo apikey save --load-for-env`

https://github.com/civo/cli/blob/99470155c7f5d03b80be9ce45832d8e6d54dc5ed/cmd/apikey/apikey_save.go#L33

Is there some other documentation that you could point me to, about these working on every command?

Reply:
No yeah I've referred the same https://github.com/civo/cli/blob/99470155c7f5d03b80be9ce45832d8e6d54dc5ed/cmd/apikey/apikey_save.go#L33
I have added the changes !!